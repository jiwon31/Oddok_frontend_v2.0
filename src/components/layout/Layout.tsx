import { Outlet } from "react-router-dom";
import Footer from "./Footer/Footer";
import Header from "./Header/Header";
import styles from "./Layout.module.css";

function Layout() {
  return (
    <div className={styles.wrap}>
      <Header />
      <div className={styles.inner}>
        <Outlet />
      </div>
      <Footer />
    </div>
  );
}

export default Layout;
