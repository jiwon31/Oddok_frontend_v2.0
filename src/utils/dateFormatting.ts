// Date 객체를 yyyy-mm-dd 형식의 문자열로 반환
export default function dateFormatting(date: Date): string {
  const year = date.getFullYear();
  const month = date.getMonth() + 1;
  const day = date.getDate();
  return `${year}-${month < 10 ? `0${month}` : month}-${day < 10 ? `0${day}` : day}`;
}
