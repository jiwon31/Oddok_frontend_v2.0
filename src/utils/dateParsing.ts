// yyyy-mm-dd 형태의 문자열을 Date 객체로 변환
export default function dateParsing(value: string): Date {
  const date = new Date(value); // 로컬 타임존이 반영된 시간
  const offset = date.getTimezoneOffset() * 60000;
  return new Date(date.getTime() + offset);
}
