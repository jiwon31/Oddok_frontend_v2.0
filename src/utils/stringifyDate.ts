export default function stringifyDate(date: Date): string {
  const offset = date.getTimezoneOffset() * 60000;
  return new Date(+date - offset).toISOString();
}
