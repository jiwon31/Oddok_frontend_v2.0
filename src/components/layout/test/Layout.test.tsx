import { Route } from "react-router-dom";
import renderer from "react-test-renderer";
import { withRecoil, withRouter } from "tests/utils";
import Layout from "../Layout";

describe("Layout", () => {
  it("renders with children component", () => {
    const component = renderer.create(
      withRecoil(
        withRouter(
          <Route path="/" element={<Layout />}>
            <Route index element={<p>children</p>} />
          </Route>,
        ),
      ),
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
