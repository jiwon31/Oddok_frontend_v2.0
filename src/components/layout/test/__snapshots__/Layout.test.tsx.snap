// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders with children component 1`] = `
<div
  className="wrap"
>
  <header
    className="header"
  >
    <a
      className="logo"
      href="/"
      onClick={[Function]}
    >
      ODDOK
    </a>
    <div
      className="pages"
    >
      <a
        className="page clicked"
        href="/"
        onClick={[Function]}
      >
        스터디룸
      </a>
      <a
        className="page false"
        href="/mypage"
        onClick={[Function]}
      >
        마이페이지
      </a>
    </div>
    <nav
      className="nav"
    >
      <a
        className="search"
        href="/search"
        onClick={[Function]}
      >
        <svg>
          search.svg
        </svg>
      </a>
      <a
        className="button"
        href="/studyroom/create"
        onClick={[Function]}
      >
        + 새 스터디 만들기
      </a>
    </nav>
  </header>
  <div
    className="inner"
  >
    <p>
      children
    </p>
  </div>
  <footer
    className="footer"
  >
    <div
      className="info"
    >
      <div
        className="title"
      >
        <h2
          className="logo"
        >
          ODDOK
        </h2>
        <p
          className="name"
        >
          실시간 화상 스터디 서비스
        </p>
      </div>
      <div
        className="contact"
      >
        <p
          className="contents"
        >
          Contact
        </p>
        <p
          className="content"
        >
          yejin013@naver.com
        </p>
      </div>
      <div
        className="team"
      >
        <p
          className="contents"
        >
          Team
        </p>
        <p
          className="content"
        >
          DOBBY @숭실대학교 소프트웨어학부
        </p>
      </div>
    </div>
    <div
      className="copyright"
    >
      <p>
        Copyright Dobby 2022, All Rights Reserved.
      </p>
    </div>
  </footer>
</div>
`;
